Add Web Push and App Badge support to MiniBrowser for testing purposes

beidson opened the pull request:
#### d65ae5ea23a9dd813a65c5686600e77bf4256113
<pre>
Add Web Push and App Badge support to MiniBrowser for testing purposes
<a href="https://bugs.webkit.org/show_bug.cgi?id=261482">https://bugs.webkit.org/show_bug.cgi?id=261482</a>
rdar://115394079

Reviewed by Andy Estes.

Fills in all the delegates and other support functionality for MiniBrowser to support Web Push from the testing service.

* Tools/MiniBrowser/MiniBrowser.xcodeproj/project.pbxproj:
* Tools/MiniBrowser/mac/AppDelegate.h:
* Tools/MiniBrowser/mac/AppDelegate.m:
(-[BrowserAppDelegate init]):
(-[BrowserAppDelegate persistentDataStore]):
(-[BrowserAppDelegate notificationPermissionsForWebsiteDataStore:]):
(-[BrowserAppDelegate userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:]):
(-[BrowserAppDelegate websiteDataStore:showNotification:]):
(+[BrowserAppDelegate currentBadge]):
(-[BrowserAppDelegate websiteDataStore:workerOrigin:updatedAppBadge:]):
(-[BrowserAppDelegate _processPendingPushMessages]):
(-[BrowserAppDelegate _handleURLEvent:withReplyEvent:]):
(-[BrowserAppDelegate awakeFromNib]):
(-[BrowserAppDelegate defaultConfiguration]):
(-[BrowserAppDelegate fetchDefaultStoreWebsiteData:]):
(-[BrowserAppDelegate fetchAndClearDefaultStoreWebsiteData:]):
(-[BrowserAppDelegate clearDefaultStoreWebsiteData:]):
(persistentDataStore): Deleted.
* Tools/MiniBrowser/mac/BrowserWindowController.h:
* Tools/MiniBrowser/mac/BrowserWindowController.m:
(-[BrowserWindowController updateTitleForBadgeChange]):
* Tools/MiniBrowser/mac/Info.plist:
* Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
(-[WK2BrowserWindowController _webView:requestNotificationPermissionForSecurityOrigin:decisionHandler:]):
(-[WK2BrowserWindowController updateTitleForBadgeChange]):
(-[WK2BrowserWindowController updateTitle:]):

Canonical link: <a href="https://commits.webkit.org/267928@main">https://commits.webkit.org/267928@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/929f9ef5ff6d16a8fa46c64ee81f8173cd2628fc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18101 "6 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18432 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18999 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19937 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/16941 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18298 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21729 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18590 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/18909 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18320 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18559 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/15752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20815 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15784 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/16505 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/23022 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16802 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/16676 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20898 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17247 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/14608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/16339 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4306 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/20700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17095 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Tools/MiniBrowser/mac/AppDelegate.h:
```diff
@@ -23,11 +23,14 @@
  */
 
+#include <UserNotifications/UNUserNotificationCenter.h>
+#include <WebKit/_WKWebsiteDataStoreDelegate.h>
```
Nit: #import

aestes commented on Tools/MiniBrowser/mac/AppDelegate.m:
```diff
@@ -73,47 +76,145 @@ - (id)init
+- (void)_processPendingPushMessages
+{
+    [[self persistentDataStore] _getPendingPushMessages:^(NSArray<NSDictionary *> *pendingPushMessages) {
+        for (NSDictionary* message in pendingPushMessages) {
```
Nit: `NSDictionary *`

beidson committed:
Add Web Push and App Badge support to MiniBrowser for testing purposes
https://bugs.webkit.org/show_bug.cgi?id=261482
rdar://115394079

Reviewed by Andy Estes.

Fills in all the delegates and other support functionality for MiniBrowser to support Web Push from the testing service.

* Tools/MiniBrowser/MiniBrowser.xcodeproj/project.pbxproj:
* Tools/MiniBrowser/mac/AppDelegate.h:
* Tools/MiniBrowser/mac/AppDelegate.m:
(-[BrowserAppDelegate init]):
(-[BrowserAppDelegate persistentDataStore]):
(-[BrowserAppDelegate notificationPermissionsForWebsiteDataStore:]):
(-[BrowserAppDelegate userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:]):
(-[BrowserAppDelegate websiteDataStore:showNotification:]):
(+[BrowserAppDelegate currentBadge]):
(-[BrowserAppDelegate websiteDataStore:workerOrigin:updatedAppBadge:]):
(-[BrowserAppDelegate _processPendingPushMessages]):
(-[BrowserAppDelegate _handleURLEvent:withReplyEvent:]):
(-[BrowserAppDelegate awakeFromNib]):
(-[BrowserAppDelegate defaultConfiguration]):
(-[BrowserAppDelegate fetchDefaultStoreWebsiteData:]):
(-[BrowserAppDelegate fetchAndClearDefaultStoreWebsiteData:]):
(-[BrowserAppDelegate clearDefaultStoreWebsiteData:]):
(persistentDataStore): Deleted.
* Tools/MiniBrowser/mac/BrowserWindowController.h:
* Tools/MiniBrowser/mac/BrowserWindowController.m:
(-[BrowserWindowController updateTitleForBadgeChange]):
* Tools/MiniBrowser/mac/Info.plist:
* Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
(-[WK2BrowserWindowController _webView:requestNotificationPermissionForSecurityOrigin:decisionHandler:]):
(-[WK2BrowserWindowController updateTitleForBadgeChange]):
(-[WK2BrowserWindowController updateTitle:]):

Canonical link: https://commits.webkit.org/267928@main
